repos/spack_repo/builtin/packages/py_mumps4py/package.py

```diff
@@ -0,0 +1,63 @@
+from spack_repo.builtin.build_systems.python import PythonPackage
+
+from spack.package import *
+
+
+class PyMumps4py(PythonPackage):
+    """Python wrapper for the MUMPS solver (MUMPS4PY)."""
+
+    homepage = "https://github.com/imadki/mumps4py"
+    url = "https://github.com/imadki/mumps4py/archive/refs/tags/1.0.0.tar.gz"
+
+    maintainers = ["williampiat3"]
+
+    version("1.0.0", sha256="634dd52a9942e88a430d6c8b519cbf6e50db77dd5f1be1cb04e5ab3f3e9da8ba")
+
+    # Python requirements
+    depends_on("python@3.5:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-pip", type="build")
+
+    # Runtime and build deps
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-cython", type="build")
+    depends_on("py-mpi4py", type=("build", "run"))
+
+    # Optional/test deps
+    depends_on("py-pytest", type="test")
+
+    # External solver
+    depends_on("mumps")
+    depends_on("cmake", type="build")
```

Comment:
This can be cleaned up using default arguments, and I think the structure makes more sense that way

```suggestion
    # build dependencies
    with default_args(type="build"):
        depends_on("cmake")
        depends_on("py-setuptools")
        depends_on("py-cython")

    # Python dependencies
    with default_args(type=("build", "run")):
        depends_on("py-numpy")
        depends_on("py-mpi4py")

    # Optional/test deps
    depends_on("py-pytest", type="test")

    # External solver
    depends_on("mumps")
```
Notes:
1. Removed dependency on `py-pip`, as that is implied by inheriting from `PythonPackage`
2. Removed `python` dependency because a python dependency is implied by inheriting from `PythonPackage`, and Spack has fully removed all python versions that are too old for this package.